conformance/lang/expressions/logical-expr/logical_and_expr.balt

```diff
@@ -0,0 +1,849 @@
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using literals.
+Labels: logical-expr, list-constructor-expr, nil-literal, float
+
+function errorFunction() {
+    boolean _ = true && (); // @error static type of each expression must be a subtype of boolean
+    boolean _ = () && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = () && (); // @error static type of each expression must be a subtype of boolean
+
+    boolean _ = true && 12; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 12 && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 12 && 13; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ = true && "text"; // @error static type of each expression must be a subtype of boolean
+    boolean _ = "text" && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = "text" && "text"; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ = true && 12.4; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 0xb34.3 && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = 0xb34.3 && 23.4; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ = true && [true, 1]; // @error static type of each expression must be a subtype of boolean
+    boolean _ = [true, 1] && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = [true, 1] && [true, 1]; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ = true && [true, false]; // @error static type of each expression must be a subtype of boolean
+    boolean _ = [true, true] && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = [true, false] && [true, true]; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using literal unions.
```

Comment:
What do we mean by "literal union"?

```suggestion
Description: Test static type of expression in logical expression via invalid references of union types.
```

Reply:
[94d5da7](https://github.com/ballerina-platform/ballerina-spec/pull/1113/commits/94d5da7d6a3c741d8d2a8cb2d22ac6910b20f776)